central/apitoken/expiration/singleton.go

```diff
@@ -0,0 +1,23 @@
+package expiration
+
+import (
+	"github.com/stackrox/rox/central/apitoken/datastore"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+var (
+	notifier Notifier
+	once     sync.Once
+)
+
+// Singleton ...
+func Singleton() Notifier {
+	if !env.PostgresDatastoreEnabled.BooleanSetting() {
+		return nil
+	}
```

Comment:
super-nit:
you could move this in the `once.Do` so subsequent calls will be a no-op and simply return nil.